crates/node/src/service/rpc/middleware.rs

```diff
@@ -275,9 +275,11 @@ async fn add_rpc_version_to_method(req: &mut hyper::Request<Body>) -> Result<(),
 
     for item in items.iter_mut() {
         if let Some(method) = item.get_mut("method").as_deref().and_then(Value::as_str) {
-            let new_method =
-                format!("starknet_{}_{}", version.name(), method.strip_prefix("starknet_").unwrap_or(method));
-
+            let new_method = if let Some((prefix, suffix)) = method.split_once('_') {
```

Comment:
Because of how `versioned_rpc` is expanded, all `jsonrpsee` macros will expand to `#[jsonrpsee::proc_macro::rpc, server, namespace = "foo"]`, and all methods will be prepended with "foo_" (this is a placeholder). As a result, this condition is always valid and `else` will never execute. Please refactor accordingly.

For context, here is the result of recursive macro expansion of `StarknetWriteRpcApiV0_7_1Server`:

```rust
#[jsonrpsee::core::__reexports::async_trait]
#[doc = "Server trait implementation for the `StarknetWriteRpcApiV0_7_1` RPC API."]
pub trait StarknetWriteRpcApiV0_7_1Server: Sized + Send + Sync + 'static {
    #[doc = " Submit a new transaction to be added to the chain"]
    async fn add_invoke_transaction(
        &self,
        invoke_transaction: BroadcastedInvokeTransaction,
    ) -> RpcResult<InvokeTransactionResult>;
    // ... 
    #[doc = "Collects all the methods and subscriptions defined in the trait and adds them into a single `RpcModule`."]
    fn into_rpc(self) -> jsonrpsee::RpcModule<Self> where {
        let mut rpc = jsonrpsee::RpcModule::new(self);
        {
            let res = rpc.register_async_method("starknet_V0_7_1_addInvokeTransaction", |params, context| async move {
                // ...
            });
            debug_assert!(
                res.is_ok(),
                "RPC macro method names should never conflict, this is a bug, please report it."
            );
        }
        // ...
}
```